CromIAM/src/main/scala/cromiam/cromwell/CromwellClient.scala

```diff
@@ -64,6 +64,33 @@ class CromwellClient(scheme: String, interface: String, port: Int, log: LoggingA
   } recoverWith {
     case e => Future.failed(CromwellConnectionFailure(e))
   }
+
+  /**
+    * Retrieve the root workflow ID for a workflow ID. This is used in case the user is inquiring about a subworkflow.
+    */
+  def getRootWorkflow(workflowId: String, user: User): Future[String] = {
+    def metadataToRootWorkflowId(metadata: WorkflowMetadata): String = {
+      import spray.json._
+      /*
+          Parse the JSON and then look for a top level field in the response object called rootWorkflowId, and return.
+          If for some reason there's not a root workflow ID (not a subworkflow and/or this is an old workflow), just
+          use the current workflow id.
+
+          This is all called from inside the context of a Future, so exceptions will be properly caught.
+      */
+      metadata.value.parseJson.asJsObject.fields.get("rootWorkflowId").map(_.toString).getOrElse(workflowId)
+    }
+
+    // FIXME: See FIXME in collectionForWorkflow
```

Comment:
yo dog could maybe be DRYer tho

Reply:
It could be, but it’s also the exact thing I mentioned as changing next in the PR description